ros/src/twist_controller/dbw_node.py

```diff
@@ -77,11 +82,8 @@ def loop(self):
             if (self.can_predict_controls()):
                 throttle, brake, steering = self.controller.control(self.target_linear_velocity,
                                                                     self.target_angular_velocity,
-                                                                    self.current_linear_velocity,
-                                                                    self.dbw_enabled)
-                if self.dbw_enabled:
-                    #   TODO - self.publish(throttle, brake, steer)
-                    pass
+                                                                    self.current_linear_velocity)
```

Comment:
Since controller stops updating if dbw_enabled is toggled OFF for a while, we may need a way to reset the I term when dbw_enabled is set back OFF -> ON.  Not sure the best way, but something to think about later.

Reply:
It would be easy enough to implement that; I could simply flush out all accumulated data from the controller(s) when the dbw_enabled flag is set to false.

Question: if the controller doesn't accumulate data if the dwb_enabled flag is false (which will happen because it's inside the `if (self.can_predict_controls())` loop on line 82, do we need to reset the controller before adding new data? I can understand that measurement consistency won't be guaranteed, but I'm not sure how much that would compromise controller output integrity once it's turned back on. What do you think?

Reply:
OK, I thought about this some more and realize that I'll need to implement this for proper PID controller usage!

Reply:
For a standard PID controller, all we should need to reset is the accumulated I term value back to 0 and reset the prev value = cur value for D term calculation.  If there is some error when the switch is enabled again, there will still be some step change in target output from P term, but I guess the actual DBW actuators and logic also have some filtering and response delay so I don’t think we need to do anything about the transition.  I’ll look into the Dataspeed docs a little to check though.

Reply:
Also, I think we should implement a feed-forward term in addition to the PID feedback to reduce the amount of feedback movement necessary. Let me know if you want to discuss it more when you’re implementing that part.

Reply:
Great, thanks for the feedback.